packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,245 @@
+import { Component, Host, h, Prop, Fragment } from '@stencil/core';
+import classnames from 'classnames';
+
+declare const window: Window &
+  typeof globalThis & {
+    dataLayer: any;
+  };
+
+@Component({
+  tag: 'va-pagination',
+  styleUrl: 'va-pagination.css',
+  shadow: true,
+})
+export class VaPagination {
+  /**
+   * Fires when a page is selected
+   */
+  @Prop() pageSelect: any;
```

Comment:
Can this be an `@Event`? Doing something like this will tell Stencil to have the `on*` handler:

```tsx
  @Event() vaChange: EventEmitter;
  // ...

  this.vaChange.emit()
```

Reply:
I'm not sure if we can do this without updating different handlers in vets-website because they expect a `number` value and don't handle the event themselves.

Reply:
This is the error if I try to prefix a prop with `on` for reference.
```
[ WARN  ]  Build Warn: ./src/components/va-pagination/va-pagination.tsx:18:11
           The @Prop() name "onPageSelect" looks like an
           event. Please use the "@Event()" decorator to
           expose events instead, not properties or
           methods.

     L17:     */
     L18:    @Prop() onPageSelect: any;
```

Reply:
What about something like this?

<details>

```diff
diff --git a/packages/storybook/stories/va-pagination.stories.jsx b/packages/storybook/stories/va-pagination.stories.jsx
index ca42d88..bebff0e 100644
--- a/packages/storybook/stories/va-pagination.stories.jsx
+++ b/packages/storybook/stories/va-pagination.stories.jsx
@@ -21,14 +21,15 @@ const Template = ({
   'show-last-page': showLastPage,
 }) => {
   const [page, setPage] = useState(currentPage);
-  const handlePageSelect = pageNumber => {
-    setPage(pageNumber);
+  const handlePageSelect = pageEvent => {
+    console.log('EVENT', pageEvent);
+    setPage(pageEvent.detail);
   };
   return (
     <VaPagination
       page={page}
       pages={pages}
-      pageSelect={handlePageSelect}
+      onPageSelect={handlePageSelect}
       showLastPage={showLastPage}
     />
   );
diff --git a/packages/web-components/src/components.d.ts b/packages/web-components/src/components.d.ts
index 9b1ec0d..dd5c91b 100644
--- a/packages/web-components/src/components.d.ts
+++ b/packages/web-components/src/components.d.ts
@@ -185,10 +185,6 @@ export namespace Components {
           * The current page number
          */
         "page": number;
-        /**
-          * Fires when a page is selected
-         */
-        "pageSelect": any;
         /**
           * The total number of pages
          */
@@ -724,13 +720,13 @@ declare namespace LocalJSX {
          */
         "maxPageListLength"?: number;
         /**
-          * The current page number
+          * Fires when a page is selected
          */
-        "page"?: number;
+        "onPageSelect"?: (event: CustomEvent<any>) => void;
         /**
-          * Fires when a page is selected
+          * The current page number
          */
-        "pageSelect"?: any;
+        "page"?: number;
         /**
           * The total number of pages
          */
diff --git a/packages/web-components/src/components/va-pagination/va-pagination.tsx b/packages/web-components/src/components/va-pagination/va-pagination.tsx
index 6dac302..d46aa01 100644
--- a/packages/web-components/src/components/va-pagination/va-pagination.tsx
+++ b/packages/web-components/src/components/va-pagination/va-pagination.tsx
@@ -1,4 +1,12 @@
-import { Component, Host, h, Prop, Fragment } from '@stencil/core';
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  Fragment,
+} from '@stencil/core';
 import classnames from 'classnames';
 
 declare const window: Window &
@@ -15,7 +23,11 @@ export class VaPagination {
   /**
    * Fires when a page is selected
    */
-  @Prop() pageSelect: any;
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  pageSelect: EventEmitter;
 
   /**
    * Fires when a page is selected if enable-analytics is true
@@ -63,7 +75,7 @@ export class VaPagination {
   };
 
   private handlePageSelect = (page, eventID) => {
-    this.pageSelect(page);
+    this.pageSelect.emit({ page });
     if (!this.enableAnalytics) return;
 
     this.handleTrackEvent({
```

</details>

Reply:
We could do this, but we would still have to update current `onPageSelect` handlers in `vets-website` to handle the event.

Reply:
I do feel like this is the best approach even if we have to make changes in `vets-website`.

Reply:
> We could do this, but we would still have to update current onPageSelect handlers in vets-website to handle the event.

I don't think that's too much work for teams to do. There might be too much variability there for this to be done automatically with the migration script, but there are only 14 instances where this happens. Manually replacing `myArg` with `myArg.detail` seems simple enough as long as we make sure there's documentation on how to handle the event.

Reply:
I'm Ok with this approach, but we need to add some documentation. I'd say go for it and I'll add some notes to the pagination migration ticket to create some extra documentation on migrating this component. We're going to have to be extra communicative about this one. It might even be good for us to work with a team on migrating their pagination component so we can provide a good example.